### ISSUE TITLE
MD030 false positive in block math

### ISSUE THREAD

**Issue by zhenghui88** [rvben/rumdl#275]
`rumdl` incorrectly reports rule `MD030` (Spaces after list markers) for content inside a block math environment. It seems to interpret lines starting with `-` inside the math block as list items.

Given the following file `test.md`:

```markdown
# Heading

$$
A = \left[
\begin{array}{c}
1 \\
-D
\end{array}
\right]
$$
```

`rumdl` reports an error on line 7 (`-D`), claiming it violates MD030.

```sh
❯ rumdl check test.md
main.md:7:2: [MD030] Spaces after list markers (Expected: 1; Actual: 0) [*]

Issues: Found 1 issues in 1 file (19ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

The check should pass without errors, as the content is within a math block (`$$ ... $$`) and should be ignored by Markdown list formatting rules.

**Comment by rvben** [comment]
Thanks @zhenghui88! Fixed in v0.0.213 - math blocks are now detected and excluded from list marker spacing checks.
